apiserver/facades/client/controller/controller.go

```diff
@@ -795,20 +795,16 @@ func (c *ControllerAPI) ModifyControllerAccess(ctx context.Context, args params.
 			continue
 		}
 
-		external := !targetUserTag.IsLocal()
 		updateArgs := access.UpdatePermissionArgs{
+			Change:  permission.AccessChange(string(arg.Action)),
+			Subject: user.NameFromTag(targetUserTag),
 			AccessSpec: permission.AccessSpec{
 				Access: permission.Access(arg.Access),
 				Target: permission.ID{
 					ObjectType: permission.Controller,
 					Key:        c.controllerTag.Id(),
 				},
 			},
-			AddUser:  true,
```

Comment:
issue: This changes the expected behavior of the facade here. If not adding a user in the permission domain update permission, you must handle the case where a user doesn't exist when ModifyControllerAccess is called and add the user.

Reply:
Same as above:
Code in 3.6 that implements this is [here](https://github.com/Aflynn50/juju/blob/5e07f2c9d041d674a524aaff0a7a599023b40ae1/state/useraccess.go#L64). The comment above mentions model users but this code is called by `AddControllerUser`.

QA:
```
➜  ~ ctrl36:test sj grant missing-user superuser
ERROR could not grant controller access: user "missing-user" does not exist locally: user "missing-user" not found
➜  ~ ctrl36:test sj grant jim@external superuser
➜  ~ ctrl36:test 
```